src/init.cpp

```diff
@@ -283,7 +305,7 @@ void Shutdown()
  */
 static void HandleSIGTERM(int)
 {
-    fRequestShutdown = true;
+    StartShutdown();
```

Comment:
Unfortunately, a signal handler is not allowed to do anything besides set a flag.
This is the only reason why we still have the polling.